client/java/transports-gcs/src/main/java/io/openlineage/client/transports/gcs/GcsTransport.java

```diff
@@ -0,0 +1,100 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.transports.gcs;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineageClientException;
+import io.openlineage.client.OpenLineageClientUtils;
+import io.openlineage.client.transports.Transport;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Optional;
+import java.util.function.Function;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class GcsTransport extends Transport {
+
+  private final Storage storage;
+  private final String bucketName;
+  private final Optional<String> fileNamePrefix;
+
+  public GcsTransport(GcsTransportConfig config) throws IOException {
+    this(buildStorage(config), config);
+  }
+
+  public GcsTransport(Storage storage, GcsTransportConfig config) {
+    this.storage = storage;
+    this.bucketName = config.getBucketName();
+    this.fileNamePrefix = Optional.ofNullable(config.getFileNamePrefix());
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull RunEvent runEvent) {
+    emitEvent(runEvent);
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull DatasetEvent datasetEvent) {
+    emitEvent(datasetEvent);
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull JobEvent jobEvent) {
+    emitEvent(jobEvent);
+  }
+
+  private static Storage buildStorage(GcsTransportConfig config) throws IOException {
+    StorageOptions.Builder builder = StorageOptions.newBuilder();
+    builder.setProjectId(config.getProjectId());
```

Comment:
Is it ok run builder each time? Not sure what is done within the builder and how expensive this operation is.